mqtt/mqtt-broker/src/broker.rs

```diff
@@ -1969,6 +2001,227 @@ pub(crate) mod tests {
         assert_matches!(sub_rx.try_recv(), Err(TryRecvError::Empty))
     }
 
+    #[tokio::test]
```

Comment:
Just as a general comment, the more I look into the broker tests, the more I believe that tests that start the broker in a separate thread should be moved to integration tests. Not right now, but probably later I will move all of them.

Reply:
I agree, the tests in broker.rs are integration tests, however I think they access some internals and it might not be possible to move them